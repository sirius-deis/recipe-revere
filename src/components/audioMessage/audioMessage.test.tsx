import { fireEvent, render, screen } from "@testing-library/react";
import AudioMessage from "./audioMessage";

describe("AudioMessage component", () => {
  it("should match snapshot", () => {
    const { container } = render(<AudioMessage src="audio.mp3" />);
    expect(container).toMatchSnapshot();
  });

  it("should render the audio player", () => {
    render(<AudioMessage src="audio.mp3" />);
    expect(screen.getByRole("audio")).toBeInTheDocument();
  });

  it("should play the audio when clicked", () => {
    const audioElement = document.createElement("audio");
    audioElement.src = "audio.mp3";
    audioElement.load();

    render(<AudioMessage src="audio.mp3" />);
    fireEvent.click(screen.getByRole("button"));

    expect(audioElement.play).toHaveBeenCalled();
  });
})