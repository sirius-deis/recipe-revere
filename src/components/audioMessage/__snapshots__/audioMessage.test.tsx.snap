// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioMessage component should match snapshot 1`] = `
<div>
  <div>
    <button
      class="btn btn-sm btn-main btn-rounded"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M48 64C21.5 64 0 85.5 0 112V400c0 26.5 21.5 48 48 48H80c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48H48zm192 0c-26.5 0-48 21.5-48 48V400c0 26.5 21.5 48 48 48h32c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48H240z"
        />
      </svg>
    </button>
    <canvas
      height="75"
      width="500"
    />
  </div>
</div>
`;
